sklearn/cluster/_kmeans.py

```diff
@@ -1061,6 +1062,10 @@ def fit_predict(self, X, y=None, sample_weight=None):
         labels : ndarray of shape (n_samples,)
             Index of the cluster each sample belongs to.
         """
+        self._backend = find_backend(self, X, y=y, sample_weight=sample_weight)
+        if self._backend is not None:
+            self._backend.fit_predict(X, y=y, sample_weight=sample_weight)
```

Comment:
We can probably find a nicer solution for this, instead of having it explicitly in the method. Maybe a decorator? Maybe via a meta-class? 

For now I thought it would be simpler to just have it explicitly in the class instead of reaching into the Python box of tricks.